app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -97,6 +98,10 @@ class SharedCallingViewModel @Inject constructor(
 
     var callState by mutableStateOf(CallState(conversationId))
 
+    var permissionPermanentlyDeniedDialogState: PermissionPermanentlyDeniedDialogState by mutableStateOf(
```

Comment:
suggestion: maybe just use `rememberVisibilityState` entirely in the compose just like we do with some other dialogs, because this view model doesn't set, update nor modify this state so it could be entirely on the UI side used in the compose and not added to the view model

Reply:
It is updating the state by calling `showPermissionPermanentlyDeniedDialog()` and `hidePermissionPermanentlyDeniedDialog()`

Reply:
I just meant that all these updates come from outside of the ViewModel because both `showPermissionPermanentlyDeniedDialog()` and `hidePermissionPermanentlyDeniedDialog()` are called from Composables and ViewModel doesn't do with this state other than just keeping it, so there's no need to keep this state in the ViewModel as it could be done entirely in the Composable just by creating
```
val permissionPermanentlyDeniedDialogState = rememberVisibilityState<PermissionPermanentlyDeniedDialogState>()
```
and then calling `show` and `dismiss` directly in Compose rather than going through the ViewModel and back to Compose 😄 
```
permissionPermanentlyDeniedDialogState.show(
    PermissionPermanentlyDeniedDialogState(
            title = R.string.app_permission_dialog_title,
            description = R.string.camera_permission_dialog_description
        )
)
```

```
permissionPermanentlyDeniedDialogState.dismiss()
```

But I won't insist, it's just a suggestion which shows another way of doing it without the need to add this state and functions to 7 different ViewModels. 😉

Reply:
We may lost dialog state in configuration change, like screen rotation that's why I am putting it on viewModel

Reply:
Changes applied